firmware/defmt-test/macros/src/lib.rs

```diff
@@ -135,13 +133,29 @@ fn tests_impl(args: TokenStream, input: TokenStream) -> parse::Result<TokenStrea
                 }
             }
 
-            Item::Use(u) => {
-                imports.push(u);
+            Item::Const(t) => untouched_tokens.push(Box::new(t)),
+            Item::Enum(t) => untouched_tokens.push(Box::new(t)),
+            Item::ExternCrate(t) => untouched_tokens.push(Box::new(t)),
+            Item::ForeignMod(t) => untouched_tokens.push(Box::new(t)),
+            Item::Impl(t) => untouched_tokens.push(Box::new(t)),
+            Item::Macro(t) => untouched_tokens.push(Box::new(t)),
+            Item::Macro2(t) => untouched_tokens.push(Box::new(t)),
+            Item::Mod(t) => untouched_tokens.push(Box::new(t)),
+            Item::Static(t) => untouched_tokens.push(Box::new(t)),
+            Item::Struct(t) => untouched_tokens.push(Box::new(t)),
+            Item::Trait(t) => untouched_tokens.push(Box::new(t)),
+            Item::TraitAlias(t) => untouched_tokens.push(Box::new(t)),
+            Item::Type(t) => untouched_tokens.push(Box::new(t)),
+            Item::Union(t) => untouched_tokens.push(Box::new(t)),
+            Item::Use(t) => untouched_tokens.push(Box::new(t)),
+            Item::Verbatim(t) => {
+                untouched_tokens.push(Box::new(t));
```

Comment:
I think you could use a wildcard match here and accumulate `syn::Item`s instead of `Box<dyn ToTokens>`

Reply:
Haha, this is a much simpler solution. :grinning: 
Was underestimating the cleverness of the `quote!` macro.